akka-actor-typed-tests/src/test/scala/akka/actor/typed/eventstream/EventStreamSpec.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.eventstream
+
+import akka.actor.testkit.typed.scaladsl.{ ScalaTestWithActorTestKit, TestProbe }
+import org.scalatest.WordSpecLike
+
+class EventStreamSpec extends ScalaTestWithActorTestKit with WordSpecLike {
+
+  import EventStreamSpec._
+
+  "system event stream".can {
+    val eventObjListener: TestProbe[EventObj.type] = testKit.createTestProbe()
+    val eventClassListener: TestProbe[EventClass] = testKit.createTestProbe()
+
+    "register subscribers" in {
+      testKit.system.eventStream ! EventStream.Subscribe(eventObjListener.ref)
+      testKit.system.eventStream ! EventStream.Subscribe(eventClassListener.ref)
+    }
+
+    "accept published events" in {
+      testKit.system.eventStream ! EventStream.Publish(EventObj)
+    }
+    "dispatch events to subscribers of that type" in {
+      eventObjListener.expectMessage(EventObj)
+      eventClassListener.expectNoMessage()
+      testKit.system.eventStream ! EventStream.Publish(EventClass())
+      eventClassListener.expectMessage(EventClass())
+      eventObjListener.expectNoMessage()
+    }
```

Comment:
The logic for type filtering is not in the added logic but for completeness, please add a test that covers listening for a supertype and publishing an instance of a subtype and the other way around (which shouldn't get a message)